src/lib.rs

```diff
@@ -259,3 +271,38 @@ pub fn make_game() -> Result<Game, JsValue> {
         players,
     })
 }
+
+pub fn start_websocket() -> Result<(), JsValue> {
+    let ws = WebSocket::new("ws://localhost:3012")?;
+    let onmessage_callback = Closure::wrap(Box::new(move |e: MessageEvent| {
+        // Starting with assuming text messages. Can make efficient later (bson?).
+        if let Ok(txt) = e.data().dyn_into::<js_sys::JsString>() {
+            console_log!("message from network: {:?}", txt);
+        } else {
+            console_log!("non-string message received! {:?}", e.data());
+        }
+    }) as Box<dyn FnMut(MessageEvent)>);
```

Comment:
:question: :weary: :question:

Reply:
`Box<dyn something>` means "This is a value that implements the trait `something`. Because we don't know the specific type, we don't know how big it is, so it needs to live on the heap and we'll hold onto a pointer to it (the Box). For most purposes we can treat it like it's some a generic `something`"

Looks like FnMut is this: https://doc.rust-lang.org/std/ops/trait.FnMut.html

All that to say though, this is just copy-pasta for me right now. Will revisit it if and when necessary